### PR TITLE
RFC: ceph-daemon unit tests

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by liewegas** [comment]
Semi-related question: should we avoid the py2-isms so that ceph-daemon can work on py2 hosts?

**Comment by sebastian-philipp** [comment]
> Semi-related question: should we avoid the py3-isms so that ceph-daemon can work on py2 hosts?

If we want to support Nautilus at some point. If we stay on Octopus+, we should be fine with Python 3.

File: src/ceph-daemon [hunk 1]
```diff
@@ -1576,31 +1576,32 @@ try:
 except NameError:
     av = sys.argv[1:]
 
-args = parser.parse_args(av)
+if __name__ != 'nosetests':
```
Comment (sebastian-philipp) [inline src/ceph-daemon]: :-1: for this quirk.

**Comment by liewegas** [comment]
> > Semi-related question: should we avoid the py3-isms so that ceph-daemon can work on py2 hosts?
> 
> If we want to support Nautilus at some point. If we stay on Octopus+, we should be fine with Python 3.

ceph-daemon only runs outside our container, so this is a matter of which OSs we want to support to *host* the containers.  Notably, I'm not seeing python3 installed by default on centos7.  Are any of these py2/3 changes bad form for python3?  The ``print(...),`` in particular seems super weird. @alfredodeza ?

File: src/ceph-daemon (hunk 1 above)
Comment (liewegas) [reply src/ceph-daemon]: yeah i don't like it either. ultimately this should be weighed against the pain of shipping around and/or a user running a modularized ceph-daemon.

File: src/test/test_ceph_daemon.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env nosetests
```
Comment (alfredodeza) [inline src/test/test_ceph_daemon.py]: it is surprising to see nosetests being added to more places in the codebase. That project is no longer maintained (it hasn't received updates in a while). I don't think using it is needed at all, and pytest would work fine.

**Pull request closed without merging**